plugins/balancedLevelProvider.go

```diff
@@ -30,6 +30,12 @@ type balancedLevelProvider struct {
 
 	// precomputed before construction
 	randGen *rand.Rand
+
+	//keeps the starting balances from the previous run to check if anything has changed
```

Comment:
comments explaining the use of variables should be on the right side of the declaration (like for the variables above)

In general for comments in the code:
as a convention it should have a space after `//` and should begin with a lowercase character